src/structs/channels.rs

```diff
@@ -1337,9 +1357,24 @@ impl Attachment {
       proxy_url: String::from(""),
       height: None,
       width: None,
-      ephemeral: None
+      ephemeral: None,
+      duration_secs: None,
+      waveform: None,
+      flags: None
     }
   }
+
+  /// Sets the duration of the file in seconds
+  pub fn set_duration_secs(mut self, duration_secs: f64) -> Self {
```

Comment:
Uploading a file is done through the [`File`](https://github.com/tonkku107/slashook/blob/c58ff0b1e6d4fbbb0288605986ea57f18708e3ff/src/structs/utils.rs#L31) struct and an attachment is only created if necessary at the [rest multipart handler](https://github.com/tonkku107/slashook/blob/c58ff0b1e6d4fbbb0288605986ea57f18708e3ff/src/rest/mod.rs#L69) or [webhook multipart handler](https://github.com/tonkku107/slashook/blob/c58ff0b1e6d4fbbb0288605986ea57f18708e3ff/src/webhook/multipart.rs#L32).   
These functions (and fields) should also be added for the `File` struct as you'd only be able to potentially use these `Attachment` ones for editing (need to test if discord allows editing these fields).